File: ios/source/api/Response.swift
import Foundation

struct Response: Codable {
    var project: Project?
    var projects: [ProjectInfo]?
    var playbackState: PlaybackState?
    var waveform: WaveformResponse?
    var progress: Progress?
    var upload: UploadAck?
    var error: String?
    var exportResponse: ExportResponse?
    var importResponse: ImportResponse?
}

struct WaveformResponse: Codable {
    var sampleId: Id
    var waveformData: WaveformData
}

struct UploadAck: Codable {
    var uploadId: Id
}

struct ExportResponse: Codable {
    var projectId: Id
    var data: Data
    var moreComing: Bool
}

struct ImportResponse: Codable {
    var projectId: Id
}

File: editor/src/api/response.ts
import {Project} from '../model/project';
import {ProjectInfo} from '../model/project-info';
import {PlaybackState} from '../model/playback-state';
import {Waveform} from '../model/waveform';
import {Progress} from '../model/progress';

interface UploadAck {
  uploadId: string;
}

export interface Response {
  project?: Project;
  playbackState?: PlaybackState;
  progress?: Progress;
  error?: string;
  projects?: ProjectInfo[];
  upload?: UploadAck;
  waveform?: Waveform;
}
